realm/realm-library/src/objectServer/java/io/realm/internal/objectstore/OsMongoCollection.java

```diff
@@ -73,14 +90,18 @@ public long getNativeFinalizerPtr() {
     }
 
     public Long count() {
-        return count(null);
+        return countInternal(new Document(), null);
+    }
+
+    public Long count(final Bson filter) {
+        return countInternal(filter, null);
     }
 
-    public Long count(@Nullable final Bson filter) {
-        return count(filter, null);
+    public Long count(final Bson filter, final CountOptions options) {
+        return countInternal(filter, options);
     }
 
-    public Long count(@Nullable final Bson filter, @Nullable final CountOptions options) {
+    private Long countInternal(final Bson filter, @Nullable final CountOptions options) {
```

Comment:
Did you intentionally remove the public possibility to call count with `options` without `filter`?

Reply:
It was a mistake, it never was part of the original API.